### ISSUE TITLE
[expo-sensors] Motion & Fitness permission requested on app reload

### ISSUE THREAD

**Issue by joeporpeglia** [expo/expo#29760]
### Minimal reproducible example

https://github.com/joeporpeglia/sensors-permission-repro

### What platform(s) does this occur on?

iOS

### Did you reproduce this issue in a development build?

Yes

### Summary

When expo-sensors is installed it will automatically request the Motion & Fitness permission in the following scenarios:
- Reloading a development build
- Reloading a production app after installing an expo update

Expected behavior: this permission is only requested when the app calls an api that needs it.

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.3.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 18.17.1 - ~/.nvm/versions/node/v18.17.1/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v18.17.1/bin/yarn
      npm: 9.6.7 - ~/.nvm/versions/node/v18.17.1/bin/npm
      Watchman: 2023.09.04.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.14.3 - /Users/joeporpeglia/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
      Android SDK:
        API Levels: 23, 24, 26, 29, 31, 33, 34
        Build Tools: 28.0.3, 30.0.2, 30.0.3, 31.0.0, 33.0.0, 33.0.1, 34.0.0
        System Images: android-31 | Google APIs ARM 64 v8a, android-31 | Google Play ARM 64 v8a, android-33 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2023.2 AI-232.10300.40.2321.11567975
      Xcode: 15.0/15A240d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~51.0.14 => 51.0.14
      expo-router: ~3.5.16 => 3.5.16
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.74.2 => 0.74.2
      react-native-web: ~0.19.10 => 0.19.12
    Expo Workflow: bare

### Expo Doctor Diagnostics

✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check dependencies for packages that should not be installed directly
✔ Check npm/ yarn versions
✔ Check for common project setup issues
✔ Check for issues with metro config
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check that packages match versions required by installed Expo SDK
✔ Check native tooling versions
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that native modules use compatible support package versions for installed Expo SDK

Didn't find any issues with the project!

**Comment by ariboren** [comment]
This issue is blocking us from using Expo Updates in prod

**Comment by joeporpeglia** [comment]
If you don't need the pedometer module, then #29845 provides a workaround for this issue.

**Comment by joeporpeglia** [comment]
Following up on the workaround I posted above: this fixes prod builds, but development builds still crash on reload and when opening the dev build home screen. This makes it impossible to switch between a local dev server and an expo update without scanning a QR code.

**Comment by joeporpeglia** [comment]
After further investigation I see a couple notable things:
- An app built with `macos-sonoma-14.5-xcode-15.4` will crash on reload. Building with `macos-sonoma-14.4-xcode-15.3` doesn't. I'm using this as a temporary workaround.
- The barometer module is using the `CMSensorRecorder` API ([see here](https://github.com/expo/expo/blob/dd4b3a52bd635dcd1bd2968c433b1525038971f6/packages/expo-sensors/ios/BarometerModule.swift#L30)). That API requires `NSMotionUsageDescription`. Setting `motionPermission: false` in the `expo-sensors` config plugin will disable the pedometer module, but it doesn't disable the barometer module. It's still unclear how this code path is reached when reloading the app though.

**Comment by juliayojji** [comment]
Same issue here!
How can I disable ask Motion & Fitness permissions and don't have app crash ?

**Comment by joeporpeglia** [comment]
@juliayojji I ended up using `patch-package` to strip all the barometer module code. Not a proper fix, but you could try that if you don't need the barometer.

**Comment by joeporpeglia** [comment]
@mfkrause yup here's the patch for expo-sensors@13.0.9. Note: I still have `motionPermission: false` set in my app config.

```patch
diff --git a/ios/BarometerModule.swift b/ios/BarometerModule.swift
index 1e4dc32b9cf63ecc6d2b398a8544f215d3126d68..305539491989dcfc3bc6af03e00e9e56129ebfab 100644
--- a/ios/BarometerModule.swift
+++ b/ios/BarometerModule.swift
@@ -23,33 +23,12 @@ public final class BarometerModule: Module {
     }
 
     OnStartObserving {
-      if CMAltimeter.authorizationStatus() == .notDetermined {
-        if #available(iOS 17.4, *) {
-          // There's a bug in iOS 17.4 where the motion permissions popup won't display
-          // even when the NSMotionUsageDescription is in the plist while using the altimeter.
-          CMSensorRecorder().recordAccelerometer(forDuration: 0.1)
-        }
-      }
-
-      altimeter.startRelativeAltitudeUpdates(to: operationQueue) { [weak self] data, _ in
-        guard let data else {
-          return
-        }
-        self?.sendEvent(EVENT_BAROMETER_DID_UPDATE, [
-          // Given pressure needs to be converted from kPa to hPa
-          "pressure": data.pressure.doubleValue * 10.0,
-          "relativeAltitude": data.relativeAltitude.doubleValue,
-          "timestamp": data.timestamp
-        ])
-      }
     }
 
     OnStopObserving {
-      altimeter.stopRelativeAltitudeUpdates()
     }
 
     OnDestroy {
-      altimeter.stopRelativeAltitudeUpdates()
     }
   }
 }
```

**Comment by doug-shontz** [comment]
I am going to go the "hack route" with patch-package, thanks for the patch!  Any update on this ticket from the expo side?  We are seeing this as well and it seems it's been around for half a year now...

**Comment by Ankkiyy** [comment]
Having the Same issue Permission is not Prompting Yet. in 2025 too.

**Comment by vanenshi** [comment]
This problem is still exist

**Comment by paulxuca** [comment]
Still an issue. package patch above worked

**Comment by EdwinVanRooij** [comment]
Issue still exists for me
